drafts/signals-everywhere.md

```diff
@@ -0,0 +1,167 @@
+
+Released back in 2010, Knockout presented a reactivity concept they called "observables".
+[Knockout describes them](https://knockoutjs.com/documentation/observables.html) as **"special JavaScript objects that can notify subscribers about changes, and can automatically detect dependencies"**.
+The examples might look surprisingly familiar for their age:
+
+```js
+function AppViewModel() {
+    this.firstName = ko.observable('Bob');
+    this.lastName = ko.observable('Smith');
+    this.fullName = ko.pureComputed(function() {
+        return this.firstName() + " " + this.lastName();
+    }, this);
+}
+```
+
+Compare this to Signals in Solid:
+
+```js
+function App() {
+    const [firstName, setFirstName] = createSignal('Bob');
+    const [lastName, setLastName] = createSignal('Smith');
+    const fullName = createMemo(() => `${firstName()} ${lastName()}`);
+}
+```
+
+If we only look at the surface, it seems like the progress of web development in over a decade has been insignificant.
+Which of course is not true.
+But using fine-grained reactivity is certainly a comeback – and it's a strong one.
+Aside from React, all major frameworks have introduced Signals or a similar concept in the last few years.
+
+As Ryan Carniato described in his article [The Quest for ReactiveScript](https://dev.to/this-is-learning/the-quest-for-reactivescript-3ka3) in 2021, there are commonly three types of reactivity:
+
+- Reactive State: Signal, Observable, Ref
+- Derived Values: Memo, Computed
+- Side Effects: Effect, Watch, Reaction, Autorun
+
+The following table is an overview of how the implementation of these types is called in their respective framework.
+
+| Framework | Signals since | Reactive State | Derived Values | Side Effect |
+|-|-|-|-|-|
+| Knockout | 2010 | observable | pureComputed | computed |
+| Svelte | 2016 | writable/readable | derived | – (reactive statements) |
+| Solid | 2018 | createSignal | createMemo | createEffect |
+| Vue | 2020 | ref/reactive | computed | watch/watchEffect |
+| Preact | 2022 | signal | computed | effect |
+| Qwik | 2022 | useSignal/useStore | useComputed$/useResource$ | useTask$ |
+| Angular | 2023 | signal | computed | effect |
+| Svelte 5 | 2024 | $state | $derived | $effect |
+
+Solid made Signals the talk of the town and has done the web community a big favour.
+For the first time, we have a robust, coherent, somewhat standardised approach for reactive user interfaces across multiple frameworks.
+The web is converging a little.
+**Signals have become transferrable knowledge**, making it easier and faster to learn a different technology.
+Which brings us – finally – to the question: what are they?
+
+## Signals explained
+
+The core idea is unchanged from Knockout's definition of observables: "special objects, which notify subscribers and can detect dependencies".
+A Signal gives you a getter and a setter, while other state solutions (like React's `useState`) give you the value and a setter.
```

Comment:
idk if "gives you" is nice here. I generally try to avoid you in a manner like that
maybe you could use something like: A Signal returns a getter and a setter, while other state solutions (like React's `useState`) return the value and a setter.